docs: Revisions to documentation, A380X referencing, file structure consolidation, and optimised naming conventions

CameronCarlyon opened the pull request:
<!-- ⚠⚠ Do not delete this pull request template! ⚠⚠ -->
<!-- Pull requests that do not follow this template are likely to be ignored. -->

<!-- Add the issues this PR fixes here. If no issues are related to this PR, then this line can be removed. -->
<!-- Add further issues with a full "Fixes #[issue_no]" line to ensure GitHub closes each one when the PR is merged. -->
Fixes #10337

## Summary of Changes
<!-- Please provide a summary of changes for this pull request, ensuring all changes are explained. -->

- Updated file name from Contributing.md to CONTRIBUTING.md and Resources.md to RESOURCES.md, in-line with FlyByWire naming conventions.

- Consolidated general development documentation into to .github folder.

- Updated CONTRIBUTING.md document to more thoroughly include references to the A380X, with minor tweaks to sentence structure throughout document to enhance readability.

- Updated README.md to ensure correct hyperlink attributes.

- Included change in CHANGELOG.md (not sure if this is necessary given the scope of the changes, feel free to remove)

## Screenshots (if necessary)
<!-- If your PR includes visual changes, screenshots from before and after your change should always be included. -->
<!-- Please make your best efforts to provide useful before and after screenshots. They should match camera angle, zoom, size, time of day, etc. -->

## References
<!-- You should be making changes based on some kind of a reference (manuals, videos, IRL photos). P3D/xplane/fsx references will only be accepted if we believe that one cannot reasonably obtain a better source. Please post screenshots of the references you used. Ask around in the discord for how to find references for what you are working on. Exceptions will probably be made for IRL A320 pilots and engineers. -->

<!-- If you are making a pull request related to the MCDU, please make sure you are ONLY referencing the Honeywell Pegasus Step 1A (Rev 0), 2009 edition manual. -->
<!-- If you do not have this manual, please ask on our discord for assistance -->

## Additional context
<!-- Add any other context about the pull request here. -->

Additional hyperlink references may need to be updated by an admin, such as these notices:

<img width="929" height="87" alt="image" src="https://github.com/user-attachments/assets/9c34cd05-2b5b-4c9b-b8ce-adf4b74fa6e7" />

<!-- You may optionally provide your discord username, so that we may contact you directly about the issue. -->
Discord username (if different from GitHub):
YouBaconMeCrazy

## Testing instructions
<!-- Detail how this PR should be tested by QA. Try to list important items that need checking, either directly changed by this PR or that could be affected by it -->

Links may be tested in the README.md file by clicking through and ensuring there is not a NoPage error.

<!-- DO NOT DELETE THIS -->
## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo**, **flybywire-aircraft-a380-842 (4K)** or **flybywire-aircraft-a380-842 (8K)** download link at the bottom of the page

CameronCarlyon committed:
Renamed to CONTRIBUTING.md and updated to reflect A380X

- Updated file name from Contributing.md to CONTRIBUTING.md, in-line with FlyByWire naming conventions.

- Updated document to more thoroughly include references to the A380X, closing issue #10337.

- Minor tweaks to sentence structure throughout document to enhance readability.

CameronCarlyon committed:
Updated CONTRIBUTING.md links in README.md

CameronCarlyon committed:
Updated CHANGELOG.md to reflect CONTRIBUTING.md change

evroon commented on .github/CONTRIBUTING.md:
```diff
@@ -39,13 +39,13 @@ If you are using WSL, ensure that the `Vmmem` process is not memory limited. At
-If you're comfortable contributing to Open Source projects on GitHub please ensure you read our expectations for issue tracking, feature proposals and pull requests.
+If you're comfortable contributing to open source projects on GitHub please ensure you read our expectations for issue tracking, feature proposals and pull requests.
 
 If you're looking for tools and tips to help you develop, see [Development Resources](../fbw-a32nx/docs/resources.md).
```
I noticed this link is broken and came across this PR. Could we fix this link? Or should I create a new issue for that?

I think it's quite important because this is the starting point for new developers

evroon replied:
I opened this pr instead: https://github.com/flybywiresim/aircraft/pull/10348

CameronCarlyon replied:
Thanks for the heads up, it's very strange as the link appears to work in VSCode, but not on the GitHub website. I have updated the branch to move the file to the .github folder as it is a general development document and not linked only to the A32NX project. I have also updated the file name to follow the standard project naming convention. As mentioned in your PR, the content of CONTRIBUTING.md could probably do with being updated as much of it is outdated.

CameronCarlyon committed:
Renamed and moved RESOURCES.md document to .github

Changed filename to UPPERCASE format, inline with established documentation naming convention.

Reference in CONTRIBUTING.md updated.

CameronCarlyon committed:
Update CHANGELOG.md

Updated CHANGELOG to include broader documentation changes.

CameronCarlyon commented:
Hey @tracernz, the RESOURCES.md and CONTRIBUTING.md files have been moved from './fbw-a32nx/docs' to './.github' thus are technically new files. This is because they are general development documentation and not applicable only to the A32NX project. Would this still be problematic with this in mind? I will revert the changelog file.

CameronCarlyon committed:
Reverted CHANGELOG.md

CameronCarlyon committed:
Merge branch 'master' into CONTRIBUTING-revision
